public/app/features/alerting/unified/components/rule-editor/LabelsField.tsx

```diff
@@ -23,94 +71,209 @@ const LabelsField: FC<Props> = ({ className }) => {
   } = useFormContext<RuleFormValues>();
 
   const labels = watch('labels');
-  const { fields, append, remove } = useFieldArray({ control, name: 'labels' });
+  const { fields, remove, append } = useFieldArray({ control, name: 'labels' });
 
-  const getDistinctDropdownOptions = (
-    fields: Array<FieldArrayWithId<RuleFormValues, 'labels', 'id'>>,
-    type: AlertLabelDropdownProps['type']
-  ) =>
-    [...new Set(fields.map((field) => field[type]))]
-      .filter((field) => Boolean(field))
-      .map((field) => ({ label: field, value: field }));
+  const dataSourceName = watch('dataSourceName');
+
+  const labelsByKey = useGetCustomLabels(dataSourceName || RuleFormType.grafana);
```

Comment:
can we relay on this condition? are we sure to have dataSourceName in the form in case of not having a grafana managed alert rule?

Reply:
Yes, I tried both in grafana managed alert rules and external alert managers. The datasource name is present in both cases. But I will do as you suggested and send it as prop from the parent component.